File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/Delphi/ViewBandedFixedDemo/ViewBandedFixedMain.pas
unit ViewBandedFixedMain;

interface

uses
  Windows, Messages, Classes, Graphics, Controls, Forms,
  Dialogs, cxGridLevel, cxControls, cxGridCustomView,
  cxGridCustomTableView, cxGridTableView, cxGridDBTableView, cxGrid, DB,
  cxGridBandedTableView, cxGridDBBandedTableView, cxStyles,
  cxGridCustomPopupMenu, cxGridPopupMenu, ViewBandedFixedMainData, StdCtrls,
  Menus, ActnList, ImgList, cxLookAndFeels, ComCtrls,
  cxCustomData, cxGraphics, cxFilter, cxData, cxEdit, cxClasses,
  cxDataStorage, cxDBData, cxCalc, cxDBLookupComboBox, cxLookAndFeelPainters,
  BaseForm;

type
  TViewBandedFixedDemoMainForm = class(TfmBaseForm)
    glUserslSchedule: TcxGridLevel;
    cxGrid: TcxGrid;
    btvUsersSchedule: TcxGridDBBandedTableView;
    miOptions: TMenuItem;
    miShowBandsHeaders: TMenuItem;
    miShowIndicator: TMenuItem;
    miShowColumnsHeaders: TMenuItem;
    miMultiSelect: TMenuItem;
    btvUsersScheduleSUNDAY: TcxGridDBBandedColumn;
    btvUsersScheduleMONDAY: TcxGridDBBandedColumn;
    btvUsersScheduleTUESDAY: TcxGridDBBandedColumn;
    btvUsersScheduleTHURSDAY: TcxGridDBBandedColumn;
    btvUsersScheduleSATURDAY: TcxGridDBBandedColumn;
    btvUsersScheduleFRIDAY: TcxGridDBBandedColumn;
    btvUsersScheduleRowAvg: TcxGridDBBandedColumn;
    btvUsersScheduleRowSum: TcxGridDBBandedColumn;
    btvUsersScheduleUserName: TcxGridDBBandedColumn;
    btvUsersScheduleProjectName: TcxGridDBBandedColumn;
    btvUsersScheduleWEDNESDAY: TcxGridDBBandedColumn;
    cxGridPopupMenu: TcxGridPopupMenu;
    procedure FormShow(Sender: TObject);
    procedure miShowBandsHeadersClick(Sender: TObject);
    procedure miShowIndicatorClick(Sender: TObject);
    procedure miShowColumnsHeadersClick(Sender: TObject);
    procedure miMultiSelectClick(Sender: TObject);
  end;

var
  ViewBandedFixedDemoMainForm: TViewBandedFixedDemoMainForm;

implementation

{$R *.dfm}

uses
  AboutDemoForm;

procedure TViewBandedFixedDemoMainForm.FormShow(Sender: TObject);
begin
  cxGrid.FocusedView.DataController.Groups.FullExpand;
end;

procedure TViewBandedFixedDemoMainForm.miShowBandsHeadersClick(Sender: TObject);
begin
  btvUsersSchedule.OptionsView.BandHeaders := GetMenuItemChecked(Sender);
end;

procedure TViewBandedFixedDemoMainForm.miShowIndicatorClick(Sender: TObject);
begin
  btvUsersSchedule.OptionsView.Indicator := GetMenuItemChecked(Sender);
end;

procedure TViewBandedFixedDemoMainForm.miShowColumnsHeadersClick(Sender: TObject);
begin
  btvUsersSchedule.OptionsView.Header := GetMenuItemChecked(Sender);
end;

procedure TViewBandedFixedDemoMainForm.miMultiSelectClick(Sender: TObject);
begin
  btvUsersSchedule.OptionsSelection.MultiSelect := GetMenuItemChecked(Sender);
end;

end.

File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/CBuilder/ViewBandedFixedDemo/ViewBandedFixedMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#include "shellapi.hpp"
#pragma hdrstop

#include "ViewBandedFixedMain.h"
#include "AboutDemoForm.h"
#include "ViewBandedFixedMainData.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxClasses"
#pragma link "cxControls"
#pragma link "cxCustomData"
#pragma link "cxData"
#pragma link "cxEdit"
#pragma link "cxFilter"
#pragma link "cxGraphics"
#pragma link "cxGrid"
#pragma link "cxGridBandedTableView"
#pragma link "cxGridCustomPopupMenu"
#pragma link "cxGridCustomTableView"
#pragma link "cxGridCustomView"
#pragma link "cxGridDBBandedTableView"
#pragma link "cxGridLevel"
#pragma link "cxGridPopupMenu"
#pragma link "cxGridTableView"
#pragma link "cxStyles"
#pragma link "cxLookAndFeels"
#pragma resource "*.dfm"
TViewBandedFixedDemoMainForm *ViewBandedFixedDemoMainForm;
//---------------------------------------------------------------------------
__fastcall TViewBandedFixedDemoMainForm::TViewBandedFixedDemoMainForm(TComponent* Owner)
  : TfmBaseForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TViewBandedFixedDemoMainForm::FormShow(TObject *Sender)
{
  ((TcxGridTableView*)cxGrid->FocusedView)->DataController->Groups->FullExpand();
}
//---------------------------------------------------------------------------

void __fastcall TViewBandedFixedDemoMainForm::miShowBandsHeadersClick(TObject *Sender)
{
   btvUsersSchedule->OptionsView->BandHeaders = ((TMenuItem*)Sender)->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TViewBandedFixedDemoMainForm::miShowIndicatorClick(TObject *Sender)
{
  btvUsersSchedule->OptionsView->Indicator = ((TMenuItem*)Sender)->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TViewBandedFixedDemoMainForm::miMultiSelectClick(TObject *Sender)
{
  btvUsersSchedule->OptionsSelection->MultiSelect = ((TMenuItem*)Sender)->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TViewBandedFixedDemoMainForm::miShowColumnsHeadersClick(TObject *Sender)
{
  btvUsersSchedule->OptionsView->Header = ((TMenuItem*)Sender)->Checked;
}
//---------------------------------------------------------------------------


